sqlglot/expressions.py

```diff
@@ -1641,6 +1683,11 @@ def join(
             on = and_(*ensure_list(on), dialect=dialect, **opts)
             join.set("on", on)
 
+        if using:
+            if isinstance(using, str):
```

Comment:
what if it's a list of strings?, ensure list should be first

Reply:
Can we also accept a list of expressions? Is that a case we should consider?

Reply:
I guess someone could pass an Identifier or a list of Identifiers, so yeah?